### PR TITLE
Allow compare/select on int4 data

### PR DESCRIPTION
**Pull request by sergey-kozub**
A simple HLO with kCompare and kSelect opcodes doesn't work for int4 types.
This PR fixes the issue.

### CODE REVIEW

**Review by akuegel — APPROVED** [review]
Thank you!

File: xla/service/layout_normalization_test.cc [hunk 1]
```diff
@@ -922,5 +922,21 @@ ENTRY main.17 {
...
+
+ENTRY main {
+  a = s4[10]{0:E(4)} parameter(0)
+  b = s4[10]{0:E(4)} parameter(1)
+  ROOT out = compare(a, b), direction=EQ
+}
+)";
+
+  CheckLayoutNormalization(hlo, R"(
+// CHECK: pred[10]{0} compare({{.*}})
```
Comment (akuegel) [inline xla/service/layout_normalization_test.cc:937]: It seems this CHECK line is wrong. We get this output:

%compare = pred[10]{0:E(4)} compare(%bitcast, %bitcast.2), direction=EQ

And as far as I understand, this is also the intended output.

Comment (sergey-kozub) [reply xla/service/layout_normalization_test.cc:937]: Without the changes in this PR, it'd produce "pred[10]{0:E(4)}", which is wrong (the PRED type cannot have element_size_in_bits layout attribute set).

Comment (sergey-kozub) [reply xla/service/layout_normalization_test.cc:937]: Are you running the test with the PR changes applied?

Comment (akuegel) [reply xla/service/layout_normalization_test.cc:937]: Or if it is not the intended output, then I guess shape_inference.cc needs another adjustment, as MakeCompareHlo will use it to derive the output layout.

Comment (sergey-kozub) [reply xla/service/layout_normalization_test.cc:937]: It looks I split my commit incorrectly, and some changes are missing. Fixing now.

Comment (sergey-kozub) [reply xla/service/layout_normalization_test.cc:937]: Added the missing change. Sorry for the inconvenience.

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**